pkgs/applications/misc/authenticator/default.nix

```diff
@@ -0,0 +1,98 @@
+{ lib
+, stdenv
+, fetchFromGitLab
+, fetchpatch
+, appstream-glib
+, desktop-file-utils
+, meson
+, ninja
+, pkg-config
+, python3
+, rustPlatform
+, wrapGAppsHook
+, gdk-pixbuf
+, glib
+, gst_all_1
+, gtk4
+, libadwaita
+, openssl
+, sqlite
+, wayland
+, zbar
+}:
+
+stdenv.mkDerivation rec {
+  pname = "authenticator";
+  version = "4.0.3";
+
+  src = fetchFromGitLab {
+    domain = "gitlab.gnome.org";
+    owner = "World";
+    repo = "Authenticator";
+    rev = version;
+    sha256 = "0fvs76f3fm5pxn7wg6sjbqpgip5w2j7xrh4siasdcl2bx6vsld8b";
+  };
+
+  cargoDeps = rustPlatform.fetchCargoTarball {
+    inherit src;
+    name = "${pname}-${version}";
+    sha256 = "1s97jyszxf24rs3ni11phiyvmp1wm8sicb0rh1jgwz4bn1cnakx4";
+  };
+
+  postPatch = ''
+    patchShebangs build-aux
+  '';
+
+  nativeBuildInputs = [
+    appstream-glib
+    desktop-file-utils
+    meson
+    ninja
+    pkg-config
+    python3
+    wrapGAppsHook
+  ] ++ (with rustPlatform; [
+    cargoSetupHook
+    rust.cargo
+    rust.rustc
+  ]);
+
+  buildInputs = [
+    gdk-pixbuf
+    glib
+    gst_all_1.gstreamer
+    gst_all_1.gst-plugins-base
+    # See https://gitlab.gnome.org/World/Authenticator/-/blob/master/build-aux/com.belmoussaoui.Authenticator.Devel.json
+    (gst_all_1.gst-plugins-good.overrideAttrs (old: {
+      patches = [
+        #(fetchpatch {
+        #  url = "https://gitlab.gnome.org/World/Authenticator/-/raw/master/build-aux/767.patch";
+        #  sha256 = "1g3zkfs248p8wvrvplwrl38vylqsafv6vapfr1nj5kg7ndfrgilf";
+        #})
+        ./767.patch
```

Comment:
Even though these are exactly the same, `fetchpatch` fails with
```
can't find file to patch at input line 807
Perhaps you used the wrong -p or --strip option?
The text leading up to this was:
--------------------------
|--- a/ext/gtk/gtkconfig.h
|+++ b/ext/gtk/gtkconfig.h
--------------------------
File to patch:
Skip this patch? [y]
Skipping patch.
1 out of 1 hunk ignored
```
Does `fetchpatch` not support creating files?

Reply:
maybe try to reproduce it with a minimal example and create an issue for it, if it's a general issue

debugging that can be a different task than this pr

Reply:
`fetchpatch` tends to break patches that create new files. But this patch is a static file so you can just use `fetchurl`.

Reply:
Actually, `"${src}/build-aux/767.patch"` might even work.

Reply:
> Actually, `"${src}/build-aux/767.patch"` might even work.

It works and I really like it. But is it preferable even though it will cause another rebuild whenever authenticator is updated?